### ISSUE TITLE
Design service for helping build responsive components

### ISSUE THREAD

**Issue by hansl** [angular/material2#45]
We should have something that works both in headless rendering and browsers.

The solution should be something along those lines (pseudo code):

``` typescript
@Injectable()
export class MdLayoutService {
  onLayoutChange: EventEmitter<MdLayoutEnum> = new EventEmitter<MdLayoutEnum>();
  get currentLayout(): MdLayoutEnum {
    return this.layout_;
  }

  constructor(private layout_: MdLayoutEnum) {
    if (!headless) {  // <-- Tricky part
      window.matchMedia('...').addListener((mql) => this.onLayoutChange.emit(...));
    }
  }
};
```

You can then provide an `MdLayoutEnum` constant in headless browsing (or if you want to fix it in the browser or unit tests).

TODO: 
- [ ] Design Doc.

**Comment by christophechevalier** [comment]
Many questions have been opened and close referencing this one.

I'm sorry to ask for ETA but is there any update on this topic ?

md-media is very handy and we need that to handle responsive in your app :)

Thanks

**Comment by grizzm0** [comment]
I guess this can be closed?

For anyone else looking for this it's part of @angular/flex-layout.
See: https://github.com/angular/flex-layout/wiki/Adaptive-Layouts#core-directives--responsive-features

**Comment by josephperrott** [comment]
Closed with #6772
